src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ReceivedZipFileRepositoryTest.java

```diff
@@ -0,0 +1,358 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.entity;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ReceivedZipFile;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ReceivedZipFileItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ReceivedZipFileRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.common.Classification;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Classification.EXCEPTION;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.DocumentType.OTHER;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.ZIPFILE_PROCESSING_STARTED;
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest
+@ExtendWith(SpringExtension.class)
+public class ReceivedZipFileRepositoryTest {
+    private static final Logger log = LoggerFactory.getLogger(ReceivedZipFileRepositoryTest.class);
+
+    @Autowired
+    private ReceivedZipFileRepository reportRepo;
+    @Autowired
+    private ProcessEventRepository eventRepo;
+    @Autowired
+    private EnvelopeRepository envelopeRepo;
+    @Autowired
+    private ScannableItemRepository scannableItemRepo;
+    @Autowired
+    private PaymentRepository paymentRepo;
+
+    @Test
+    void should_return_single_event_if_no_envelope() {
+        // given
+        Instant createdDate1 = Instant.parse("2019-02-14T14:15:23.456Z");
+        Instant createdDate2 = Instant.parse("2019-02-15T14:15:23.456Z");
+        Instant createdDate3 = Instant.parse("2019-02-16T14:15:23.456Z");
```

Comment:
i guess it would be more readable with `createDate1.plusDays(1)` and `createDate1.plusDays(2)`